src/sysfiles.c

```diff
@@ -753,6 +749,22 @@ SYS_SY_BUF syBuf [256];
 
 SYS_SY_BUFFER syBuffers [ 32];
 
+// Mark a member of syBuf as unused
+void SyMarkBufUnused(Int i)
+{
+    GAP_ASSERT(i >= 0 && i < 256);
+    syBuf[i].fp = -1;
+}
+
+// There is no explicit method to mark syBufs as used,
+// they are marked as used when created.
+
+// Check if a member of syBuf is in use
+static Int SyBufInUse(Int i)
+{
+    GAP_ASSERT(i >= 0 && i < 256);
+    return syBuf[i].fp != -1;
+}
```

Comment:
(Hmm, I tried to comment earlier with my phone, but it seems those comments didn't make it?)
It's a bit odd that marking a buffer unused is public, but checking whether it is in use is private. BTW, `src/streams.c` also could in theory use `SyBufInUse` when it does `if (syBuf[ifid].bufno >= 0) {`. So perhaps it should be public, too?

Alternatively, the code using `SyMarkBufUnused` in `src/system.c` could be moved into a new `InitSysBuf` function in `sysfiles.c` which is then call from `system.c`

Reply:
I don't mind make syBufInUse public, I just didn't need it anywhere outside this file, so made it static.

`syBuf[ifid].bufno >= 0` isn't actually checking if `ifid` is in use -- `SyFtell` earlier in the function already called SyBufInUse and returned -1 if it failed. It's checking if the `syBuf` has a `syBuffer` attached to it, which is an entirely different problem (yes, the SyBuf / SyBuffer naming might be unfortunate).